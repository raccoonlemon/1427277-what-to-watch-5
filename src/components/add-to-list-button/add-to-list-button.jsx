import PropTypes from 'prop-types';
import React from "react";
import {connect} from 'react-redux';
import {Path} from '../../const';
import {redirectToRoute} from '../../store/action';
import {updateIsFilmFavorite} from '../../store/api-actions';
import {selectIsUserLogged} from '../../store/selectors';

export const AddToListButton = (props)=>{
  const {id, isFavorite, onClickAction, redirectAction, isUserLogged} = props;

  return (<button
    className="btn btn--list movie-card__button"
    type="button"
    onClick = {()=>{
      if (!isUserLogged) {
        redirectAction();
      } else {
        onClickAction(id, !isFavorite);
      }
    }}>
    <svg viewBox="0 0 19 20" width="19" height="20">
      <use xlinkHref={isFavorite ? `#in-list` : `#add`}></use>
    </svg>
    <span>My list</span>
  </button>);
};

AddToListButton.propTypes = {
  id: PropTypes.string.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  onClickAction: PropTypes.func.isRequired,
  redirectAction: PropTypes.func.isRequired,
  isUserLogged: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  isUserLogged: selectIsUserLogged(state)
});

const mapDispatchToProps = (dispatch) => ({
  onClickAction(id, isFavorite) {
    dispatch(updateIsFilmFavorite(id, isFavorite));
  },
  redirectAction() {
    dispatch(redirectToRoute(Path.SIGN_IN));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(AddToListButton);
