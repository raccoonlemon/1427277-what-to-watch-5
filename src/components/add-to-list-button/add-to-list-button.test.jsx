import React from "react";
import renderer from "react-test-renderer";
import {AddToListButton} from "./add-to-list-button";

const noop = () => {};
const id = `1`;
describe(`<AddToListButton> renders correctly`, () => {
  it(`film is in the list`, () => {
    const tree = renderer
    .create(<AddToListButton
      id ={id}
      isFavorite = {true}
      isUserLogged = {true}
      onClickAction={noop}
      redirectAction={noop}/>);

    expect(tree).toMatchSnapshot();
  });

  it(`film is not in the list`, () => {
    const tree = renderer
    .create(<AddToListButton
      id ={id}
      isFavorite = {false}
      isUserLogged = {true}
      onClickAction={noop}
      redirectAction={noop}/>);

    expect(tree).toMatchSnapshot();
  });

});
