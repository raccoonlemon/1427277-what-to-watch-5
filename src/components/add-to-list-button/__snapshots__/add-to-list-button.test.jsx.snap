// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToListButton> renders correctly film is in the list 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`<AddToListButton> renders correctly film is not in the list 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;
